deeppavlov/models/ranking/siamese_model.py

```diff
@@ -71,17 +70,22 @@ def __call__(self, batch: Iterable[List[np.ndarray]]) -> Union[np.ndarray, List[
             yp = self._predict_on_batch(b)
             y_pred += list(yp)
         y_pred = np.asarray(y_pred)
-        if len(responses) > 1:
-            y_pred = np.reshape(y_pred, (j, len(responses)))
+        # reshape to [batch_size, n_responses] if needed (n_responses > 1)
+        y_pred = np.reshape(y_pred, (j, n_responses)) if n_responses > 1 else y_pred
         return y_pred
 
     def reset(self) -> None:
         pass
 
-    def _train_on_batch(self, batch: List[np.ndarray], y: List[int]) -> float:
+    def _append_sample_to_batch_buffer(self, sample: List, buf: Union[List[List[np.ndarray]], List[Tuple[np.ndarray]]]):
```

Comment:
Does this function reduce the amount of code?

Reply:
no changes to `_train_on_batch`:
https://github.com/deepmipt/DeepPavlov/blob/99e7d890b6d366337ac97477bf3d31c048370561/deeppavlov/models/ranking/siamese_model.py#L85